mmdet3d/models/roi_heads/bbox_heads/multi_group_head.py

```diff
@@ -0,0 +1,951 @@
+    """
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 kernel_size=3,
+                 deform_groups=4):
+        super().__init__(in_channels, out_channels, kernel_size, deform_groups)
+        offset_channels = kernel_size * kernel_size * 2
+        self.conv_offset = nn.Conv2d(
+            in_channels, deform_groups * offset_channels, 1, bias=True)
+
+    def forward(self, x):
+        """Forward function for CenterPointFeatureAdaption.
+
+        Args:
+            x (torch.Tensor): Input feature with the shape of [B, 64, W, H].
+
+        Returns:
+            torch.Tensor: Output feature with the same shape of input feature.
+        """
+        offset = self.conv_offset(x)
+        x = self.relu(self.conv_adaption(x, offset))
+        return x
+
+
+@HEADS.register_module()
+class SepHead(nn.Module):
+    """SepHead for CenterHead.
+
+    Args:
+        in_channels (int): Input channels for conv_layer.
+        heads (dict): Conv information.
+        head_conv (int): Output channels.
+            Default: 64.
+        final_kernal (int): Kernal size for the last conv layer.
+            Deafult: 1.
+        init_bias (float): Initial bias. Default: -2.19.
+        conv_cfg (dict): Config of conv layer.
+            Default: dict(type='Conv2d')
+        norm_cfg (dict): Config of norm layer.
+            Default: dict(type='BN2d').
+    """
+
+    def __init__(self,
+                 in_channels,
+                 heads,
+                 head_conv=64,
+                 final_kernel=1,
+                 init_bias=-2.19,
+                 conv_cfg=dict(type='Conv2d'),
+                 norm_cfg=dict(type='BN2d'),
+                 **kwargs):
+        super(SepHead, self).__init__(**kwargs)
+
+        self.heads = heads
+        for head in self.heads:
+            classes, num_conv = self.heads[head]
+
+            fc_layers = []
+            for i in range(num_conv - 1):
+                fc_layers.append(
+                    build_conv_layer(
+                        conv_cfg,
+                        in_channels,
+                        head_conv,
+                        kernel_size=final_kernel,
+                        stride=1,
+                        padding=final_kernel // 2,
+                        bias=True))
+                if norm_cfg:
+                    fc_layers.append(build_norm_layer(norm_cfg, head_conv)[1])
+                fc_layers.append(nn.ReLU(inplace=True))
+
+            fc_layers.append(
+                build_conv_layer(
+                    conv_cfg,
+                    head_conv,
+                    classes,
+                    kernel_size=final_kernel,
+                    stride=1,
+                    padding=final_kernel // 2,
+                    bias=True))
+            fc = nn.Sequential(*fc_layers)
+            if 'hm' in head:
+                fc[-1].bias.data.fill_(init_bias)
+            else:
```

Comment:
Move the weight initialization part into `self.init_weights()`, do the same thing for all nn modules.